Sources/CoreMetrics/Metrics.swift

```diff
@@ -488,6 +580,62 @@ public protocol MetricsFactory {
     func destroyTimer(_ handler: TimerHandler)
 }
 
+/// Wraps a CounterHandler, adding support for incrementing by floating point values by storing an accumulated floating point value and recoding increments to the underlying CounterHandler after crossing integer boundaries.
+private class AccumulatingRoundingFloatingPointCounter: FloatingPointCounterHandler {
+    var rawValue: Double = 0
+    var previousValue: Int64 = 0
+    var counterHandler: CounterHandler
+
+    init(counterHandler: CounterHandler) {
+        self.counterHandler = counterHandler
+    }
+
+    func increment(by amount: Double) {
+        // FIXME: does this need a lock?
```

Comment:
or use an atomic

Reply:
Wouldn't the entire function body need to be atomic, otherwise concurrent increments could mess-up the internal state